### PR TITLE
Implement Face Detection ios example using mediapipe tasks visions

### PR DESCRIPTION
**Pull request by st-tuanmai**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

### Checklist
Please ensure the following items are complete before submitting a pull request:

- [x] My code follows the code style of the project.
- [ ] I have updated the documentation (if applicable).
- [x] I have added tests to cover my changes.

### Type of Change
Please check the relevant option below:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update (non-breaking change which updates documentation)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Screenshots
IMG_7458
IMG_7459
IMG_7460

### Additional Notes
Add any additional information or context about the pull request here.

### CODE REVIEW

**Comment by PaulTR** [comment]
I believe the face detection model supports GPU. Can we add that dropdown to test?

**Comment by st-tuanmai** [comment]
> I believe the face detection model supports GPU. Can we add that dropdown to test?

I can't find GPU/CPU option in FaceDetectorOptions and BaseOptions

**Comment by PaulTR** [comment]
Alright confirmed that it isn't exposed yet. Thanks! Running through review now.

File: examples/face_detector/ios/FaceDetector/Helper/Define.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// MARK: Define default constants
+enum DefaultConstants {
+  static let minSuppressionThreshold: Float = 0.5
+  static let minDetectionConfidence: Float = 0.5
+  static let modelPath: String? = Bundle.main.path(forResource: "blaze_face_short_range", ofType: "tflite")
+}
+
+/// A result from the `FaceDetectorHelper`.
+struct FaceDetectorHelperResult {
```
Comment (PaulTR) [inline examples/face_detector/ios/FaceDetector/Helper/Define.swift:26]: This should be in the helper, yeah? We should follow the same changes/advice that Khanh mentioned in the image classification PR.

Comment (PaulTR) [reply examples/face_detector/ios/FaceDetector/Helper/Define.swift:26]: Yeah let's move this under the helper.

Comment (PaulTR) [reply examples/face_detector/ios/FaceDetector/Helper/Define.swift:26]: Also rename it to ResultBundle

File: examples/face_detector/ios/FaceDetector/Helper/Define.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import MediaPipeTasksVision
+
+// MARK: Define default constants
+enum DefaultConstants {
```
Comment (PaulTR) [inline examples/face_detector/ios/FaceDetector/Helper/Define.swift:19]: All of these values should be moved so we can remove this file.

File: examples/face_detector/ios/FaceDetector/Helper/FaceDetectorHelper.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  /**
+   This method return FaceDetectorResult and infrenceTime when receive an image
+   **/
+  func detect(image: UIImage) -> FaceDetectorHelperResult? {
+    guard let mpImage = try? MPImage(uiImage: image) else { return nil }
+    do {
+      let startDate = Date()
+      let result = try faceDetector?.detect(image: mpImage)
+      let inferenceTime = Date().timeIntervalSince(startDate) * 1000
```
Comment (PaulTR) [inline examples/face_detector/ios/FaceDetector/Helper/FaceDetectorHelper.swift:44]: Same as in image classification, you don't need to throttle the camera. MediaPipe will handle flow control for you.

File: examples/face_detector/ios/FaceDetector/Helper/FaceDetectorHelper.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    } catch {
+      print(error)
+      return nil
+    }
+  }
+
+  /**
+   This method return FaceDetectorResult and infrenceTime when receive videoFrame
+   **/
+  func detect(videoFrame: CVPixelBuffer, timeStamps: Int) -> FaceDetectorHelperResult? {
```
Comment (PaulTR) [inline examples/face_detector/ios/FaceDetector/Helper/FaceDetectorHelper.swift:55]: Can we rename this to detectVideo()? It'll make it easier for devs to find when they're looking through our samples.

**Pull request merged**